Try to deploy device-plugin in the edge node but edge_core crashes

<!-- Please use this template while reporting a bug and provide as much info as possible. Thanks!-->

**What happened**:
Try to deploy device-plugin in the edge node to make it support GPU but edge_core crashes instantly.
Process:
1. run "sudo kubeedge init --docker-version=18.09.6" on the server.
2. run "sudo kubeedge join --server=202.38.86.171:8080 --docker-version=18.09.6" on the edge node.
3. run "kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/1.0.0-beta/nvidia-device-plugin.yml".

The following information was revealed from "edge_core.log" showing the edge_core crashed:
2019-05-22 10:17:21.938 +08:00 [0;34mINFO[0m containers/container_manager.go:318 no container id for pod nvidia-device-plugin-daemonset-42pwh, need start container
2019-05-22 10:17:21.938 +08:00 [0;34mINFO[0m containers/container_manager.go:607 computePodActions got {ContainersToStart:[0] ContainersRestartCount:0 ContainersToKill:map[]} for pod "nvidia-device-plugin-daemonset-42pwh"
2019-05-22 10:17:21.939 +08:00 [0;34mINFO[0m containers/container_manager.go:624 create container nvidia-device-plugin-ctr in pod nvidia-device-plugin-daemonset-42pwh
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1646ad8]

goroutine 116 [running]:
github.com/kubeedge/kubeedge/edge/pkg/edged/containers.EnableHostUserNamespace(...)
	/home/root1/go/src/github.com/kubeedge/kubeedge/edge/pkg/edged/containers/helpers.go:65
github.com/kubeedge/kubeedge/edge/pkg/edged/containers.(*containerManager).StartPod(0xc0006361b0, 0xc000a2d500, 0xc000c00180, 0xc000db9d40, 0x0)
	/home/root1/go/src/github.com/kubeedge/kubeedge/edge/pkg/edged/containers/container_manager.go:680 +0x1548
github.com/kubeedge/kubeedge/edge/pkg/edged.(*edged).consumePodAddition(0xc00003c960, 0xc000b81fa0, 0x38, 0xbf314a23abe49dd3)
	/home/root1/go/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:571 +0x665
github.com/kubeedge/kubeedge/edge/pkg/edged.(*edged).podAddWorkerRun.func1(0xc00003c960, 0x3)
	/home/root1/go/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:488 +0x536
created by github.com/kubeedge/kubeedge/edge/pkg/edged.(*edged).podAddWorkerRun
	/home/root1/go/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:467 +0x95

**What you expected to happen**:
I hope the device-plugin run successfully in the edge node.

**How to reproduce it (as minimally and precisely as possible)**:
1. run "sudo kubeedge init --docker-version=18.09.6" on the server.
2. run "sudo kubeedge join --server=202.38.86.171:8080 --docker-version=18.09.6" on the edge node.
3. run "kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/1.0.0-beta/nvidia-device-plugin.yml".

**Anything else we need to know?**:
I had tried running "nvidia-device-plugin" successfully in k8s deployed with the same environment.

**Environment**:
- KubeEdge version: v0.3.0-beta.0
- Hardware configuration: 
- OS (e.g. from /etc/os-release):ubuntu 16.04
- Kernel (e.g. `uname -a`):4.15.0-50-generic
- Others:

--- a/edge/pkg/edged/containers/helpers.go
+++ b/edge/pkg/edged/containers/helpers.go
@@ -62,7 +62,7 @@ func GenerateEnvList(envs []v1.EnvVar) (result []string) {
 
 //EnableHostUserNamespace checks security to enable host user namespace
 func EnableHostUserNamespace(pod *v1.Pod) bool {
-	if pod.Spec.Containers[0].SecurityContext != nil && *pod.Spec.Containers[0].SecurityContext.Privileged {
+	if pod.Spec.Containers[0].SecurityContext != nil && pod.Spec.Containers[0].SecurityContext.Privileged != nil && *pod.Spec.Containers[0].SecurityContext.Privileged {
 		return true
 	}
 	return false

